src/app/_utils/fileUtils.ts

```diff
@@ -0,0 +1,34 @@
+import fs from 'fs'
+import path from 'path'
+
+import { notFound } from 'next/navigation'
+
+import matter from 'gray-matter'
+
+export function getFilePath(directoryPath: string, slug: string): string {
+  return path.join(directoryPath, `${slug}.md`)
+}
+
+export function handleFileNotFound(filePath: string): void {
+  console.error(`File not found: ${filePath}`)
+  notFound()
+}
+
+export function readFileContents(filePath: string): string {
+  return fs.readFileSync(filePath, 'utf8')
+}
+
+export function parseMarkdown(fileContents: string): {
+  data: object
+  content: string
+} {
+  return matter(fileContents)
+}
+
+export function getFilenamesFromDirectory(directory: string): string[] {
+  return fs.readdirSync(directory)
+}
+
+export function extractSlugFromFilename(filename: string): string {
+  return path.parse(filename).name
+}
```

Comment:
I'm not sure we need all of these helper functions.

`readFileContents` & `getFilenamesFromDirectory` encapsulate very little logic. It feels simpler to me - less cognitive overhead - to just use the file system methods here.

I'm tempted to say the same for `parseMarkdown` and `handleFileNotFound`, even though the `console.error` can be useful for debugging.

We can discuss it further if you want, maybe there's a way to group and reduce the number of functions here.